Orange/base.py

```diff
@@ -229,6 +229,9 @@ def predict(self, X):
         if type(self).predict_storage is Model.predict_storage:
             raise TypeError("Descendants of Model must overload method predict")
         else:
+            warnings.warn("Automatic fallback to predict_storage will be removed in 3.36. "
+                          "All models need to implement predict.",
+                          OrangeDeprecationWarning)
```

Comment:
What do you expect people to do, when they see this warning?
If the model did not implement a predict before, they could just copy the 4 lines below into their predict which of course would solve nothing?
If I understand the original problem correctly, the Table should be created with a copy of X. If that is the fix, I think it is better to put it here one time and avoid people maybe implementing predict incorrectly in their own forced implementations.

Reply:
I currently just do not see any good reasons that anyone would directly implement `predict_storage`. Most likely they should implement `predict` and I'd guess that in most of cases they would not need to construct a table.

The only exception I can imagine is that they want to call sublearner's `__call__`.

Looking towards counterexamples.

Reply:
Main counterexamples are when working with non- numpy/Table data or some additional information not found in X/Y/metas.
- SqlTable used fit/predict storage because you can offload processing to the sql server (if you use X, it downloads a sample). Although we are not actively working on that, I like that the architecture supports the option to.
- I see that Survival analysis uses fit_storage, because it needs more than X&Y, but not for predicting, where X is enough? @JakaKokosar
- You are developing Dask tables, maybe we will want to adjust and optimize some learner in the future that will use some DaskTable methods/caching? (but more likely in this case we could just have an `if` in standard predict to check the type of X and use dask functions directly).
- We have Corpus and Timeseries classes that extend Table and maybe some models will want to use them and the numpy arrays will not be enough

Bottom line, if you are questioning why "anyone would directly implement predict_storage", then the discussion should be if we want to remove that method completely - and as long as everything in Orange is based on Tables and we support its extensions, I would lean towards leaving this support and not rely exclusively on numpy arrays. At least until we switch to pandas, then we should re-evaluate :)

Reply:
Thanks for a list where this could go wrong. I agree. Let's not deprecate.